air/src/air.rs

```diff
@@ -108,6 +109,10 @@ pub trait AirBuilder: Sized {
     }
 }
 
+pub trait AirBuilderWithPublicValues: AirBuilder {
+    fn public_values(&self) -> Vec<Self::F>;
```

Comment:
Let's return `&[Self::F]` so the impl doesn't have to clone

Reply:
changed